src/components/menu-bar/menu-bar.jsx

```diff
@@ -152,6 +154,24 @@ class MenuBar extends React.Component {
             this.props.onRequestCloseEdit();
         };
     }
+    handleUpdateProject (updateFun) {
+        return () => {
+            this.props.onRequestCloseFile();
+            this.setState({projectSaveInProgress: true},
+                () => {
+                    updateFun().then(() => {
+                        this.setState({projectSaveInProgress: false});
```

Comment:
In the long run this will need error handling, but that's probably a separate task. It may be a little confusing not to have a confirmation that the project actually saved.

Reply:
Errors and save confirmation should all be handled by the upcoming messaging system.